todo-list-backend/src/index.ts

```diff
@@ -1,33 +1,208 @@
+
     res
-    .setHeader('Access-Control-Allow-Origin', '*')
-    .status(200)
-    .json(responseContent);
+        .status(200)
+        .send(JSON.stringify(newTodo))
+        ; console.log(JSON.stringify(newTodo)); return
 });
 
+app.post("/remove-todo", express.json(), (req: Request, res: Response) => {
+    console.log("removing todo")
+
+    if (req.body == undefined) {
+        res
+            .status(400)
+            .send("request body undefined.  sending response code 400")
+            ; console.log("request body undefined.  sending response code 400"); return
+    }
+
+    if (req.body.id == undefined) {
+        res
+            .status(400)
+            .send("task id undefined.  sending response code 400")
+            ; console.log("task id undefined.  sending response code 400"); return
+    }
+
+    let id = req.body.id
+
+    console.log("removing task id:: " + id)
+
+    let idx = todoItems.findIndex(function (item) { return item.id == id })
+
+    if (idx == -1) {
+        res
+            .status(400)
+            .send("no element in list of requested index")
+            ; console.log("no element in list of requested index"); return
+    }
+
+    let removedElement = todoItems.splice(idx, 1)[0]
+
+    res
+        .status(200)
+        .send(JSON.stringify(removedElement))
+});
+
+app.post("/tag-task-as-complete", express.json(), (req: Request, res: Response) => {
+    console.log("tagging task as complete")
+    const request = req.body;
+    const id = request.id
+    const strict = request.strict;
+    let result
+    try {
+        result = changeCompletionStatusHandler(id, strict, true)
+    } catch (e) {
+        if (e instanceof Error) {
+            res
+                .status(400)
+                .send(e.message)
+                ;
+        } else {
+            res
+                .status(500)
+                .send("unknown backend error")
+        }
+        return
+    }
+    res
+        .setHeader("Access-Control-Allow-Origin", "*")
+        .status(200)
+        .send(JSON.stringify(result))
+    console.log('task tagged as complete')
+});
+
+app.post("/tag-task-as-incomplete", express.json(), (req: Request, res: Response) => {
+    console.log("tagging task as incomplete")
+    const request = req.body;
+    const id = request.id
+    const strict = request.strict;
+    let result
+    try {
+        result = changeCompletionStatusHandler(id, strict, false)
+    } catch (e) {
+        if (e instanceof Error) {
+            res
+                .status(400)
+                .send(e.message)
+                ;
+        } else {
+            res
+                .status(500)
+                .send("unknown backend error")
+        }
+        return
+    }
+    res
+        .setHeader("Access-Control-Allow-Origin", "*")
+        .status(200)
+        .send(JSON.stringify(result))
+});
+
+function changeCompletionStatusHandler(id: number, strict: boolean, newStatus: boolean) {
```

Comment:
One bit of general advice - it's best to only add functionality if you know you'll need it. For example, a strict mode does add extra functionality to your app, but it also makes it more complicated, and can contribute to making changes in the future more tricky. I think an app like this is the perfect place to mess around with stuff like this, since it's all experimentation, but just wanted to share my experience.